### PR TITLE
Add tests to verify value type hash code behaviour

### PR DESCRIPTION
**Pull request by AditiS11**
- Add test for hash code on value types with null-restricted fields 
covering flattened and non-flattened cases.
- Add test for hash code on value types with identity fields.
- Add test to validate reading hash code from object backfill after move.
- Add test for hash code on value types with null reference fields. 
Related: #23502

### CODE REVIEW

**Review by theresa-m — APPROVED** [review]

File: test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTestClasses.java [hunk 1]
```diff
@@ -197,4 +197,20 @@ public long getL4() {
...
+	}
+
+	static value class ValueClassWithIdentityField {
+		int i;
+		IdentityClassForHashCode r;
+		ValueClassWithIdentityField(int i) {
+			this.i = i;
+			this.r = new IdentityClassForHashCode(i);
+		}
+	}
```
Comment (hangshao0) [inline test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTestClasses.java:215]: We also need a test case of VT that has different value fields but the same identity field. 
You can add another ValueClass with identity field like:
```
static value class ValueClassSubArray {
		int startFrom;  /* start index of subarray */
        int endAt; /* end index of the subarray */
		int[] Array;
}
```
You can let `int[] Array` points to a global static final array. Add a test cases for the hashcode of 2 ValueClassSubArray instances that have different start/end indexes but the same `int[] Array` field.

File: test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java [hunk 2]
```diff
@@ -2646,6 +2646,71 @@ static public void testValueClassHashCode() throws Throwable {
...
+	 */
+	@Test(priority=1, enabled = false)
+	static public void testValueTypeHashCodeAfterGC() throws Throwable {
+		Object p1 = new ValueTypePoint2D(new ValueTypeInt(1), new ValueTypeInt(2));
+		Object p2 = new ValueTypePoint2D(new ValueTypeInt(1), new ValueTypeInt(2));
+
+		int h1_beforeGC = p1.hashCode();
+		int h2_beforeGC = p2.hashCode();
+
+		System.gc();
```
Comment (hangshao0) [inline test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java]: I suggest adding `System.gc()` in all the new test cases in this PR to ensure the hashcode of the same object is consistent across GC.

Comment (AditiS11) [reply test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java]: Can I remove `testValueTypeHashCodeAfterGC` as reading the hash code from the object backfill will already be validated with the above change in the other tests.

Comment (hangshao0) [reply test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java]: > Can I remove testValueTypeHashCodeAfterGC

Yes.

**Comment by hangshao0** [comment]
Jenkins test sanity.functional,extended plinuxval jdknext

**Review by hangshao0 — APPROVED** [review]
